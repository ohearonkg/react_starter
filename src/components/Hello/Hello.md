```
<Hello name={"Sarah"} age={12} />
```
