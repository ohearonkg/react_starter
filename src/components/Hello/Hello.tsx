import * as React from "react";
import styled from "react-emotion";

export interface IHello {
  /**
   * The name of person
   * to be printed
   */
  name: string;

  /**
   * The age of the person
   * to be printed
   */
  age: number;
}

const HelloWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
`;

const ItemWrapper = styled.div`
  flex: 1 1 auto;
  text-align: center;
`;

const NameHeading = styled.h1`
  margin: 0;
  font-size: 32px;
`;

const AgeHeading = styled.h2`
  margin: 0;
  font-size: 24px;
`;

const Hello = (props: IHello) => {
  return (
    <HelloWrapper>
      <ItemWrapper>
        <NameHeading> Hello {props.name} </NameHeading>
      </ItemWrapper>

      <ItemWrapper>
        <AgeHeading> Age {props.age} </AgeHeading>
      </ItemWrapper>
    </HelloWrapper>
  );
};

export default Hello;
