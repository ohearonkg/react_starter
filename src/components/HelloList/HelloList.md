```
<HelloList hellos={[{name: "Sarah", age:12}, {name: "KG", age: 24}]} />
```