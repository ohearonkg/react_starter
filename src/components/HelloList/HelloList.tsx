import * as React from "react";
import styled from "react-emotion";
import Hello, { IHello } from "../Hello/Hello";

interface IHelloList {
  /**
   * A list of hello objects
   */
  hellos: IHello[];
}

const HelloListWrapper = styled.div`
  padding: 10px;
`;

const HelloList = (props: IHelloList) => {
  return (
    <HelloListWrapper>
      {props.hellos.map((hello: IHello, index: number) => (
        <Hello name={hello.name} age={hello.age} key={index} />
      ))}
    </HelloListWrapper>
  );
};

export default HelloList;
