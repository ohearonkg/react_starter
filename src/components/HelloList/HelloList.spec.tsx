import * as emotion from "emotion";
import { mount, ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import { createSerializer } from "jest-emotion";
import * as React from "react";
import Hello from "../Hello/Hello";
import HelloList from "./HelloList";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("Hello List Component", () => {
  /**
   * We create a new shallowly
   * mounted instance of our component
   * before each test
   */
  let wrapper: ShallowWrapper;
  let mounted: ReactWrapper;
  beforeEach(() => {
    wrapper = shallow(
      <HelloList
        hellos={[
          {
            age: 24,
            name: "Kevin"
          },
          {
            age: 22,
            name: "Sarah"
          }
        ]}
      />
    );
    mounted = mount(
      <HelloList
        hellos={[
          {
            age: 24,
            name: "Kevin"
          },
          {
            age: 22,
            name: "Sarah"
          }
        ]}
      />
    );
  });

  it("Should match its snapshot", () => {
    expect(toJson(mounted)).toMatchSnapshot();
  });

  it("Should render a <Hello /> item for each element within its hellos prop", () => {
    expect(wrapper.find(Hello)).toHaveLength(2);
  });
});
