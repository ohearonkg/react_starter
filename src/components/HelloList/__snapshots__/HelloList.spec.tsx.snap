// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello List Component Should match its snapshot 1`] = `
.emotion-20 {
  padding: 10px;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  text-align: center;
}

.emotion-0 {
  margin: 0;
  font-size: 32px;
}

.emotion-4 {
  margin: 0;
  font-size: 24px;
}

<HelloList
  hellos={
    Array [
      Object {
        "age": 24,
        "name": "Kevin",
      },
      Object {
        "age": 22,
        "name": "Sarah",
      },
    ]
  }
>
  <HelloListWrapper>
    <div
      className="emotion-20 emotion-21"
    >
      <Hello
        age={24}
        key="0"
        name="Kevin"
      >
        <HelloWrapper>
          <div
            className="emotion-8 emotion-9"
          >
            <ItemWrapper>
              <div
                className="emotion-2 emotion-3"
              >
                <NameHeading>
                  <h1
                    className="emotion-0 emotion-1"
                  >
                     Hello 
                    Kevin
                     
                  </h1>
                </NameHeading>
              </div>
            </ItemWrapper>
            <ItemWrapper>
              <div
                className="emotion-2 emotion-3"
              >
                <AgeHeading>
                  <h2
                    className="emotion-4 emotion-5"
                  >
                     Age 
                    24
                     
                  </h2>
                </AgeHeading>
              </div>
            </ItemWrapper>
          </div>
        </HelloWrapper>
      </Hello>
      <Hello
        age={22}
        key="1"
        name="Sarah"
      >
        <HelloWrapper>
          <div
            className="emotion-8 emotion-9"
          >
            <ItemWrapper>
              <div
                className="emotion-2 emotion-3"
              >
                <NameHeading>
                  <h1
                    className="emotion-0 emotion-1"
                  >
                     Hello 
                    Sarah
                     
                  </h1>
                </NameHeading>
              </div>
            </ItemWrapper>
            <ItemWrapper>
              <div
                className="emotion-2 emotion-3"
              >
                <AgeHeading>
                  <h2
                    className="emotion-4 emotion-5"
                  >
                     Age 
                    22
                     
                  </h2>
                </AgeHeading>
              </div>
            </ItemWrapper>
          </div>
        </HelloWrapper>
      </Hello>
    </div>
  </HelloListWrapper>
</HelloList>
`;
